apps/server/src/modules/learnroom/common-cartridge/utils.ts

```diff
@@ -1,6 +1,34 @@
-import { ObjectId } from 'bson';
-
-export function createIdentifier(id?: string | ObjectId): string {
-	id = id ?? new ObjectId();
-	return `i${id.toString()}`;
-}
+import { Builder } from 'xml2js';
+import { CommonCartridgeVersion } from './common-cartridge.enums';
+
+export type OmitVersion<T> = Omit<T, 'version'>;
+
+export type OmitVersionAndFolder<T> = Omit<T, 'version' | 'folder'>;
+
+const xmlBuilder = new Builder({ headless: true });
+
+export function buildXmlString(obj: unknown): string {
+	return xmlBuilder.buildObject(obj);
+}
+
+export function createVersionNotSupportedError(version: CommonCartridgeVersion): Error {
+	return new Error(`Version ${version} is not supported`);
+}
+
+export function checkCommonCartridgeVersion(version: CommonCartridgeVersion): void | never {
+	const supportedVersions = [CommonCartridgeVersion.V_1_1_0, CommonCartridgeVersion.V_1_3_0];
```

Comment:
Can we extract this array from the method to the other definitions for this file?